### PR TITLE
AES CMAC - VerifyInit/Verify

### PR DESCRIPTION
**Pull request by dachalco**
Required for NTP Client

### CODE REVIEW

File: source/portable/mbedtls/core_pkcs11_mbedtls.c [hunk 1]
```diff
@@ -2370,11 +2376,11 @@ static CK_RV prvCreateRsaKey( CK_ATTRIBUTE * pxTemplate,
 }
 
 /**
- * @brief Parses attribute values for a HMAC or CMAC Key.
```
Comment (lundinc2) [inline source/portable/mbedtls/core_pkcs11_mbedtls.c:2373]: This looks like a merge error

File: source/portable/mbedtls/core_pkcs11_mbedtls.c [hunk 2]
```diff
@@ -2492,6 +2498,69 @@ static CK_RV prvCreateSHA256HMAC( CK_ATTRIBUTE * pxTemplate,
     return xResult;
 }
 
+/**
+ * @brief Parses attribute values for a RSA Key.
```
Comment (lundinc2) [inline source/portable/mbedtls/core_pkcs11_mbedtls.c:2502]: Should be a CMAC key

File: source/portable/mbedtls/core_pkcs11_mbedtls.c [hunk 3]
```diff
@@ -4277,6 +4345,107 @@ static CK_RV prvVerifyInitSHA256HMAC( P11Session_t * pxSession,
...
+                        mbedtlsLowLevelCodeOrDefault( lMbedTLSResult ) ) );
+            prvCMACCleanUp( pxSession );
+            xResult = CKR_KEY_HANDLE_INVALID;
+        }
+    }
+
+    if( xResult == CKR_OK )
+    {
+        lMbedTLSResult = mbedtls_cipher_cmac_starts( &pxSession->xCMACSecretContext,
+                                                     pucKeyData, 8 * ulKeyDataLength );
```
Comment (lundinc2) [inline source/portable/mbedtls/core_pkcs11_mbedtls.c:4406]: Can we make the 8 * ulKeyDataLength a variable? It is hard to understand what the purpose is for multiplying by 8.

File: source/portable/mbedtls/core_pkcs11_mbedtls.c [hunk 4]
```diff
@@ -4741,6 +4929,40 @@ CK_DECLARE_FUNCTION( CK_RV, C_Verify )( CK_SESSION_HANDLE hSession,
                     }
                 }
             }
+            else if( pxSessionObj->xOperationVerifyMechanism == CKM_AES_CMAC )
+            {
```
Comment (lundinc2) [inline source/portable/mbedtls/core_pkcs11_mbedtls.c:4933]: At the end of this block, no matter the result, can you set the CMAC object handle to invalid

**Review by lundinc2 — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

**Pull request merged**